test/unit/test_disco_acm.py

```diff
@@ -0,0 +1,71 @@
+TEST_CERTIFICATE_ARN_ACM_WILDCARD = "arn:aws:acm::123:wildcard"
+
+TEST_CERT = {CERT_ARN_KEY: TEST_CERTIFICATE_ARN_ACM_EXACT, DOMAIN_NAME_KEY: TEST_DOMAIN_NAME}
+TEST_WILDCARD_CERT = {CERT_ARN_KEY: TEST_CERTIFICATE_ARN_ACM_WILDCARD,
+                      DOMAIN_NAME_KEY: TEST_WILDCARD_DOMAIN_NAME}
+
+
+class DiscoACMTests(TestCase):
+    '''Test disco_acm.py'''
+
+    def setUp(self):
+        self._acm = MagicMock()
+        self.disco_acm = DiscoACM(self._acm)
+
+    def test_get_certificate_arn_exact_match(self):
+        """exact match between the host and cert work"""
+        self._acm.list_certificates.return_value = {CERT_SUMMARY_LIST_KEY: [TEST_CERT]}
+        self.assertEqual(TEST_CERTIFICATE_ARN_ACM_EXACT,
+                         self.disco_acm.get_certificate_arn(TEST_DOMAIN_NAME),
+                         'Exact matching of host domain name to cert domain needs to be fixed.')
+
+    def test_get_certificate_arn_wildcard_match(self):
+        """wildcard match between the host and cert work"""
+        self._acm.list_certificates.return_value = {CERT_SUMMARY_LIST_KEY: [TEST_WILDCARD_CERT]}
+        self.assertEqual(TEST_CERTIFICATE_ARN_ACM_WILDCARD,
+                         self.disco_acm.get_certificate_arn(TEST_DOMAIN_NAME),
+                         'Exact matching of host domain name to cert domain needs to be fixed.')
+
+    def test_get_certificate_arn_deep_match(self):
+        """a host that matches a wildcard cert domain more than one level down should return a cert"""
+        self._acm.list_certificates.return_value = {CERT_SUMMARY_LIST_KEY: [TEST_WILDCARD_CERT]}
+        self.assertEqual(TEST_CERTIFICATE_ARN_ACM_WILDCARD,
+                         self.disco_acm.get_certificate_arn(TEST_DEEP_DOMAIN_NAME),
+                         'Matching of host domain name to a nested cert subdomain needs to be fixed.')
+
+    def test_get_certificate_arn_no_match(self):
+        """host that does not match cert domains should NOT return a cert"""
+        self._acm.list_certificates.return_value = {CERT_SUMMARY_LIST_KEY: [TEST_CERT, TEST_WILDCARD_CERT]}
+        self.assertFalse(self.disco_acm.get_certificate_arn('non.existent.cert.domain'),
+                         'Matching of host domain name to cert domain is generating false positives.')
+
+    def test_get_cert_arn_match_most_specific(self):
+        """
+        test both orderings of exact and wildcard matching cert domains
+        to ensure the host domain matches the most specific cert domain in both cases
+        """
+        self._acm.list_certificates.return_value = {CERT_SUMMARY_LIST_KEY: [TEST_CERT, TEST_WILDCARD_CERT]}
+        self.assertEqual(TEST_CERTIFICATE_ARN_ACM_EXACT,
+                         self.disco_acm.get_certificate_arn(TEST_DOMAIN_NAME),
+                         'Failed to match most specific cert domain.')
+        self._acm.list_certificates.return_value = {CERT_SUMMARY_LIST_KEY: [TEST_WILDCARD_CERT, TEST_CERT]}
```

Comment:
It could, but it seemed reasonable to group them since together they prove that the more specific match will be used regardless of the order the certs are provided.